das/daser.go

```diff
@@ -229,6 +251,14 @@ func (d *DASer) catchUp(ctx context.Context, job *catchUpJob) (int64, error) {
 				// error as nil since the routine was ordered to stop
 				return height - 1, nil
 			}
+			var byzantineErr *ipld.ErrByzantine
+			if errors.As(err, &byzantineErr) {
+				err := d.fService.Broadcast(ctx, fraud.CreateBadEncodingProof(uint64(h.Height), byzantineErr))
+				if err != nil {
+					log.Errorw("fraud proof propagating failed with error ", err)
+				}
+				return height, err
+			}
```

Comment:
Now it seems even more that `sample` and `catchUp` can be DRYied. 
cc @renaynay. 

Orthogonal to the PR